audio/src/fetch.rs

```diff
@@ -328,7 +335,8 @@ impl AudioFileOpenStreaming {
             stream_loader_command_rx,
             complete_tx,
         );
-        self.session.spawn(move |_| fetcher);
```

Comment:
I'm not that familiar with the existing code so I have to ask, what is `session`? Is this the current tokio runtime? Because in `0.2` there is a global implicit runtime that you can `spawn` into, no need to carry around specific `Handle`s anymore and stuff. I see you've got some code commented out below so I assume that we're moving in that direction or?

Reply:
Since this was written back in `tokio-core` times by @plietar, `Session` was/is a ref counted object that could be cloned and passed around to spawn futures where ever needed.

With new Tokio, indeed we can use either the thread local Handle,or the implicit runtime and just `spawn` as we please and given that is the idiomatic way to do it with tokio 0.2, we'll move towards that. 

The Audio file/fetcher part of it is quite straight forward to port IMO, I find part of Core need more thought. Especially [`channel`](https://github.com/librespot-org/librespot/blob/tokio_migration/core/src/channel.rs) and [`mercury`](https://github.com/librespot-org/librespot/blob/tokio_migration/core/src/mercury/mod.rs). I am not sure we need Session to be Ref counted any more -- but I could be wrong.

Reply:
`Session` is used to hold the tokio handle, but also the "write end" of the connection to Spotify servers.
The first purpose would go away in tokio 0.2, but you still need the session to send packets.

Reply:
I think session is due for a wider rework at some point - tokio has changed a lot in recent years and also there is still that reconnection issue that is more of an architectural one that requires significant rewriting of existing logic.